src/libs/elektra/keyhelpers.c

```diff
@@ -197,10 +197,8 @@ int keyNameIsSystem (const char * name)
  */
 int keyInit (Key * key)
 {
-	if (key->meta) ksClear (key->meta);
-	memset (key, 0, sizeof (struct _Key) - sizeof (KeySet *));
-
-	if (!key->meta) key->meta = ksNew (0, KS_END);
+	memset (key, 0, sizeof (struct _Key));
+	key->meta = ksNew (0, KS_END);
```

Comment:
Is it possible to not allocate key->meta eagerly? It was discussed in #3142 to not do this.

Reply:
This code is based on the state before #3142.
If it is possible in #3142, then it should be possible here too. I will wait for #3142 to be merged an then integrate it's changes. (Same goes for any other PRs that make changes to the core)